### PR TITLE
Move tile data to Lua

### PR DESCRIPTION
**Pull request by Ruin0x11**
# Summary
- Refactor `chipm` into `core.map_chip`.
- Changed the map chip loading mechanism. Now all chips are loaded into `PicLoader` at startup regardless of atlas kind. It could be optimized if necessary to only load the chips in the current map.
- Add tinted buffer tracker so map chips can be tinted based on the color of the day.

I managed to successfully add a new tile kind using this new setup, but we'd likely need a new map format for ease of use. I was considering Tiled, but it will need custom import/export plugins to work well with the data system.

### CODE REVIEW

File: src/elona/draw.cpp [hunk 1]
```diff
@@ -706,170 +716,168 @@ void create_pcpic(int cc, bool with_equipments)
...
-                chipm(0, cnt2 * 33 + cnt + 20) = 4;
-            }
-        }
-        for (int cnt = 33; cnt < 66; ++cnt)
-        {
-            chipm(0, cnt) = 4;
+            gmode(0);
+            tinted_buffers.tint(buffer, color);
+            gmode(2, 30);
+            tinted_buffers.tint(buffer, color, true);
```
Comment (ki-foobar) [inline src/elona/draw.cpp]: In the second call of `TintedBuffers::tint()`, should argument of `color` be `snail::Color{255}`?

```cpp
// Part of map_prepare_tileset_atlas()
    set_color_mod(255 - shadow, 255 - shadow, 255 - shadow, 6);     // set color modifier
    gcopy(6, 0, 0, 33 * inf_tiles, 25 * inf_tiles, 0, 0);      // draw all map chips including feat chips with the tint.
    set_color_mod(255, 255, 255, 6);        // clear the color modifier
    gmode(2, 30);        // set blend mode and opacity.
    if (map_data.atlas_number == 0)
    {
        gcopy(6, 0, 192, 1360, 48, 0, 192);       // draw feat chips
    }
    ...
```

So in original code, feat chips were drawn twice, drawn with the same tint as other map chips, and then drawn over with alpha 30 and without tint (in original color).

File: src/elona/draw.cpp [hunk 2]
```diff
@@ -706,170 +716,168 @@ void create_pcpic(int cc, bool with_equipments)
...
-        chipm(0, 569) = 4;
-        chipm(0, 570) = 4;
-        for (int cnt = 99; cnt < 132; ++cnt)
-        {
-            chipm(0, cnt) = 7;
-        }
-        for (int cnt = 165; cnt < 198; ++cnt)
+        auto& atlas = chip_data.get_map(data.atlas);
+
+        atlas[data.id] = std::move(data);
```
Comment (ki-foobar) [inline src/elona/draw.cpp]: Judging from using `std::move` here, the passed `MapChipDB` becomes empty state after the for loop ends, is it intended behavior?  If intended, please add comment about that in either this function or declaration of `the_map_chip_db`.

**Comment by ki-foobar** [comment]
> Tiled

Tiled Map Editor is one of the best map editor for free, but I disagree with introducing it to the project becaus it is better that all tools for modding are provided by this project only. The easier to create mods is, the better. Of course I know the default map editor built-in vanilla is an unkind tool, so I'm developing yet another map editor, distributed in my twitter (#OpatosMapEditor).

**Comment by Ruin0x11** [comment]
Understood. I was also thinking about implementing the map editor in-game, but I just thought it would save some time re-implementing many of the things the map editor would need that Tiled gives for free, like setting variable properties on objects. Also, there is no need to have a dependency on Tiled's map format, since it can import/export other formats with Python plugins. (I prototyped a 1.22 importer [here](https://github.com/Ruin0x11/foobar_tiled).) If data will no longer use integer IDs then a new map format will probably be needed. It was also important that it be cross-platform/open-source. I can't seem to find the source for Opatos anywhere, and I can't seem to get it to run under Wine on my system.

File: src/elona/draw.cpp (hunk 2 above)
Comment (Ruin0x11) [reply src/elona/draw.cpp]: I removed it to be consistent with the other `init` functions.

**Comment by Ruin0x11** [comment]
Thanks for reviewing.

**Pull request merged**